com.ibm.streamsx.metrics/com.ibm.streamsx.metrics.service/MetricsIngestService.spl

```diff
@@ -0,0 +1,69 @@
+//
+// ****************************************************************************
+// * Copyright (C) 2017, International Business Machines Corporation          *
+// * All rights reserved.                                                     *
+// ****************************************************************************
+//
+
+/**
+ * A service which publishes the [com.ibm.streamsx.metrics::MetricsSource] 
+ * operator's tuples in JSON format to the 'streamsx/metrics/ingest' topic.
+ * 
+ * Output JSON Schema:
+ * {
+ *   "domainId" : string,
+ *   "instanceId" : string,
+ *   "jobId" : string,
+ *   "jobName" : string,
+ *   "resource" : string,
+ *   "peId" : long,
+ *   "operatorName" : string,
+ *   "channel" : int,
+ *   "portIndex" : int,
+ *   "connectionId" : string,
+ *   "metricType" : string,
+ *   "metricKind" : string,
+ *   "metricName" : string,
+ *   "metricValue" : long,
+ *   "lastTimeRetrieved" : long
+ * }
+ */
+ 
+namespace com.ibm.streamsx.metrics.service;
+
+use com.ibm.streamsx.metrics::* ;
+use com.ibm.streamsx.json::TupleToJSON ;
+use com.ibm.streamsx.topology.topic::Publish ;
+
+public composite MetricsIngestService
+{			
+	param
+		expression<rstring> $applicationConfigurationName ;
+
+	graph
+		/**
+		 * The MetricsSource generates a tuple for each retrieved metric.
+		 */
+		stream<Notification> MetricsSource_0 = MetricsSource()
+		{
+			param
+				applicationConfigurationName : $applicationConfigurationName ;
+		}
+
+		/**
+		 * The TupleToJSON converts the MetricType tuples into JSON.
+		 */
+		stream<rstring jsonString> TupleToJSON_0 = TupleToJSON(MetricsSource_0)
+		{
+		
+		}
+		
+		/**
+		 * The Publish publishes the metrics JSON string to the 'streamsx/metrics/ingest' topic.
+		 */
+		() as Publish_0 = Publish(TupleToJSON_0)
+		{
+			param
+				topic : "streamsx/metrics/ingest" ;
```

Comment:
E.g. think about the case where the service is enhanced to publish some calculated results, e.g. aggregates, e.g. something like `streamsx/metrics/tuples/rates`, so the base topic should clearly indicate what is being published.